scene/resources/style_box.cpp

```diff
@@ -308,111 +352,590 @@ int StyleBoxFlat::_get_additional_border_size(Margin p_margin) const {
-			color_downright.g = (border_size - i) * color_downright.g / border_size + i * bg_color.g / border_size;
-			color_downright.b = (border_size - i) * color_downright.b / border_size + i * bg_color.b / border_size;
-		}
+	_update_default_margin();
+	emit_changed();
+}
+void StyleBoxFlat::set_corner_radius(const int radius_top_left, const int radius_top_right, const int radius_botton_right, const int radius_bottom_left) {
+	corner_radius[0] = radius_top_left;
+	corner_radius[1] = radius_top_right;
+	corner_radius[2] = radius_botton_right;
+	corner_radius[3] = radius_bottom_left;
 
-		vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r.position.x, r.position.y + r.size.y - 1), Size2(r.size.x, 1)), color_downright);
-		vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r.position.x + r.size.x - 1, r.position.y), Size2(1, r.size.y)), color_downright);
+	_update_default_margin();
+	emit_changed();
+}
+void StyleBoxFlat::set_corner_radius_TL(int radius) {
 
-		vs->canvas_item_add_rect(p_canvas_item, Rect2(r.position, Size2(r.size.x, 1)), color_upleft);
-		vs->canvas_item_add_rect(p_canvas_item, Rect2(r.position, Size2(1, r.size.y)), color_upleft);
+	corner_radius[0] = radius;
 
-		r.position.x++;
-		r.position.y++;
-		r.size.x -= 2;
-		r.size.y -= 2;
-	}
+	_update_default_margin();
+	emit_changed();
+}
+void StyleBoxFlat::set_corner_radius_TR(int radius) {
+
+	corner_radius[1] = radius;
 
-	if (draw_center)
-		vs->canvas_item_add_rect(p_canvas_item, Rect2(r.position, r.size), bg_color);
+	_update_default_margin();
+	emit_changed();
+}
+void StyleBoxFlat::set_corner_radius_BR(int radius) {
 
-	Rect2i r_add = p_rect;
-	vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r_add.position.x - additional_border_size[MARGIN_LEFT], r_add.position.y - additional_border_size[MARGIN_TOP]), Size2(r_add.size.width + additional_border_size[MARGIN_LEFT] + additional_border_size[MARGIN_RIGHT], additional_border_size[MARGIN_TOP])), light_color);
-	vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r_add.position.x - additional_border_size[MARGIN_LEFT], r_add.position.y), Size2(additional_border_size[MARGIN_LEFT], r_add.size.height)), light_color);
-	vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r_add.position.x + r_add.size.width, r_add.position.y), Size2(additional_border_size[MARGIN_RIGHT], r_add.size.height)), dark_color);
-	vs->canvas_item_add_rect(p_canvas_item, Rect2(Point2i(r_add.position.x - additional_border_size[MARGIN_LEFT], r_add.position.y + r_add.size.height), Size2(r_add.size.width + additional_border_size[MARGIN_LEFT] + additional_border_size[MARGIN_RIGHT], additional_border_size[MARGIN_BOTTOM])), dark_color);
+	corner_radius[2] = radius;
+
+	_update_default_margin();
+	emit_changed();
 }
+void StyleBoxFlat::set_corner_radius_BL(int radius) {
 
+	corner_radius[3] = radius;
+
+	_update_default_margin();
+	emit_changed();
+}
 float StyleBoxFlat::get_style_margin(Margin p_margin) const {
+	//	int margin_size = border_size;
```

Comment:
Don't do that. Either drop the code, of if you *really* need to keep it around, use `#if 0 ... #endif` to disable it instead of commenting out all lines.